fog/sample-paykit/src/client_builder.rs

```diff
@@ -255,85 +234,97 @@ impl ClientBuilder {
-            let mut mr_signer_verifier = MrSignerVerifier::new(
-                signature.mrsigner().into(),
-                signature.product_id(),
-                signature.version(),
-            );
-            mr_signer_verifier
+        let mut verifier = Verifier::default();
+        verifier.debug(DEBUG_ENCLAVE);
+
+        let mut mr_enclave_verifier =
+            MrEnclaveVerifier::from(mc_consensus_enclave_measurement::sigstruct());
+        mr_enclave_verifier
+            .allow_hardening_advisories(mc_consensus_enclave_measurement::HARDENING_ADVISORIES);
+        verifier.mr_enclave(mr_enclave_verifier);
+
+        for sig in &self.consensus_sigstruct {
+            let mut mr_enclave_verifier = MrEnclaveVerifier::from(sig.clone());
+            mr_enclave_verifier
                 .allow_hardening_advisories(mc_consensus_enclave_measurement::HARDENING_ADVISORIES);
-            mr_signer_verifier
-        } else {
-            mc_consensus_enclave_measurement::get_mr_signer_verifier(None)
-        };
+            verifier.mr_enclave(mr_enclave_verifier);
+        }
+
+        log::debug!(self.logger, "Consensus Verifier: {:?}", verifier);
 
-        let mut verifier = Verifier::default();
-        verifier.debug(DEBUG_ENCLAVE).mr_signer(mr_signer_verifier);
         verifier
     }
 
-    // Get fog ingest verifier (dynamic or build time, MRSIGNER)
+    // Get fog ingest verifier (dynamic + build time, MRENCLAVE)
     fn get_fog_ingest_verifier(&self) -> Verifier {
-        let mr_signer_verifier = if let Some(signature) = self.fog_ingest_sigstruct.as_ref() {
-            let mut mr_signer_verifier = MrSignerVerifier::new(
-                signature.mrsigner().into(),
-                signature.product_id(),
-                signature.version(),
-            );
-            mr_signer_verifier.allow_hardening_advisories(
+        let mut verifier = Verifier::default();
+        verifier.debug(DEBUG_ENCLAVE);
+
+        let mut mr_enclave_verifier =
+            MrEnclaveVerifier::from(mc_fog_ingest_enclave_measurement::sigstruct());
+        mr_enclave_verifier
+            .allow_hardening_advisories(mc_fog_ingest_enclave_measurement::HARDENING_ADVISORIES);
+        verifier.mr_enclave(mr_enclave_verifier);
+
+        for sig in &self.fog_ingest_sigstruct {
+            let mut mr_enclave_verifier = MrEnclaveVerifier::from(sig.clone());
+            mr_enclave_verifier.allow_hardening_advisories(
                 mc_fog_ingest_enclave_measurement::HARDENING_ADVISORIES,
             );
-            mr_signer_verifier
-        } else {
-            mc_fog_ingest_enclave_measurement::get_mr_signer_verifier(None)
-        };
+            verifier.mr_enclave(mr_enclave_verifier);
+        }
+
+        log::debug!(self.logger, "Fog Ingest Verifier: {:?}", verifier);
 
-        let mut verifier = Verifier::default();
-        verifier.debug(DEBUG_ENCLAVE).mr_signer(mr_signer_verifier);
         verifier
```

Comment:
this logic is somewhat repetitive, it might be nice to make a helper function somewhere, but I didn't want to go too deep on refactoring right now.